autogluon/task/tabular_prediction/tabular_prediction.py

```diff
@@ -54,7 +54,7 @@ def load(output_directory, verbosity=2):
     
     @staticmethod
     def fit(train_data, label, tuning_data=None, output_directory=None, problem_type=None, eval_metric=None,
-            hyperparameter_tune=False, feature_prune=False, holdout_frac=None, num_bagging_folds=0, stack_ensemble_levels=0,
+            hyperparameter_tune=False, feature_prune=False, auto_stack=False, holdout_frac=None, num_bagging_folds=0, num_bagging_sets=None, stack_ensemble_levels=0,
```

Comment:
Perhaps it will be cleaner in the future if we expand the scope of `auto_stack` beyond stacking to automatically decide on all fit() arguments?  

One could instead move it to be the first argument after `eval_metric` and call it something like:  'optimize_args', which when = True could overwrite all the accuracy-affecting arguments including:  `hyperparameters`, `hyperparameter_tune`, `feature_prune`, `holdout_frac`,  `num_bagging_folds`, `num_bagging_sets`, `stack_ensemble_levels`, `search_strategy`, `search_options`.  Of course, for now this argument can just overwrite the stacking-related arguments.  Essentially `optimize_args` would try to optimize absolutely everything for accuracy within the allotted time_limits (if `time_limits` is off, then fit() with optimize_args=True might be expected to run much slower).

Reply:
I agree we should introduce this parameter in future, however I don't think it should be in this PR. I think auto_stack can remain as is and this new optimize_args parameter could override it once it is introduced.